### PR TITLE
Add Hypertension report to Progress tab

### PR DESCRIPTION
**Pull request by claudiovallejo**
**Story card:** [sc-7581](https://app.shortcut.com/simpledotorg/story/7581/add-new-reports-card-and-create-hypertension-report)

## Because

We will show a full hypertension report in the progress tab that includes the following data:
* Total assigned patients (current count)
* Total registered patients trend (last 6 months)
* Monthly follow-up patients trend (last 6 months)
* BP controlled trend (last 6 months)
* BP not controlled trend (last 6 months)
* Missed visits trend (last 6 months)
* Quarterly cohort reports (last 4 quarters)

## This addresses
**Original progress report home page**
image

**New progress report home page**
image

**New progress hypertension report**
image

## Test instructions
**1. Make sure the original home page works IF the Flipper flag is not enabled:**
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Make sure the "Today" card shows data for past days.
* Make sure the "Registered patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Follow-up patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Hypertension controlled" card works. Tap the chart area and make sure the month labels show. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Click the "Cohort reports" button and make sure the cohort report page shows up. When you click out of the page, make sure the home page appears.
* Make sure the "Achievements" card shows total follow-up and total registrations values that match the totals shown in the cards above. Also make sure the badges match the values. For example, if there are 27 follow-ups, there should be 1 completed badge and 4 empty badges.

**2. Make sure the new home page and hypertension report show IF the Flipper flag, `new_progress_tab` is enabled:**
* Go to Settings > Flipper and add/enable the `new_progress_tab` flag.
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Make sure the "Today" card shows data for the past days.
* Make sure the "Registered patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Follow-up patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the old "Hypertension controlled" card and "Cohort reports" button aren't showing.
* Make sure the new "Reports" card is showing and that the "Hypertension" button is active and the "Diabetes report" button is gray and says "Coming soon"
* Click the "Hypertension report" button, and scroll down the page. Make sure the charts look good. Make sure the "Patient treatment outcomes" section help icon and chart data point tooltips work and show the right information/data. Make sure the "Cohort report" bar charts work and show the tooltips correctly above each bar.

### CODE REVIEW

**Comment by dburka** [comment]
This all looks good @claudiovallejo. The tooltips are well done, the charts look simple and good. A few suggestions:
* The dates in the bar charts can't wrap. Is that intentional. I'm worried that people with large font sizes on Android will have confusing overlapping dates in the charts — it's very common to see healthcare workers with larger font sizes.
* Grey titles like "Patient treatment outcomes" could use more bottom margin. Too close to the card.
* White text on yellow is pretty low contrast in the cohort reports. Can we maybe make them yellow-dark? The white text wouldn't pass accessibility guidelines fwiw.

**Comment by claudiovallejo** [comment]
Thanks @dburka! I've implemented the changes you requested. Regarding the text/background contrast in the cohort reports, I used a web readability checker to ensure the contrast ratio passes most tests.

image

File: app/controllers/api/v3/analytics/user_analytics_controller.rb [hunk 1]
```diff
@@ -10,14 +10,17 @@ def show
...
     @total_follow_ups = Reports::MonthlyProgressComponent.new(@total_follow_ups_dimension, service: @service).total_count
     @total_registrations = Reports::MonthlyProgressComponent.new(@total_registrations_dimension, service: @service).total_count
-    if Flipper.enabled?(:new_progress_tab)
-      @period_reports_data = Reports::ReportsFakeFacilityProgressService.new(@current_facility.name).period_reports
-      @hypertension_reports_data = Reports::ReportsFakeFacilityProgressService.new(@current_facility.name).hypertension_reports
-      @diabetes_reports_data = Reports::ReportsFakeFacilityProgressService.new(@current_facility.name).diabetes_reports
+
+    @drug_stocks = DrugStock.latest_for_facilities_grouped_by_protocol_drug(current_facility, @for_end_of_month)
+    unless @drug_stocks.empty?
+      @query = DrugStocksQuery.new(facilities: [current_facility],
```
Comment (harimohanraj89) [inline app/controllers/api/v3/analytics/user_analytics_controller.rb:21]: Let's call this `@drug_stocks_query`. There's a lot going on in this page so`@query` wouldn't make it very clear what it is.

**Comment by claudiovallejo** [comment]
> Minor feedback, feel free to address and ship.
> 
> The view components and data sources look clean ✨
> 
> I've only looked at the code, I'll put the onus of testing on you to verify before we ship.

Thanks for the comments @harimohanraj89! I've resolved all of them :)

File: app/views/api/v3/analytics/user_analytics/_progress_cohort_report.html.erb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<div id="progress-cohort-report" class="progress-body" style="display: none;">
```
[the remark this answers is not in the document]
Comment (claudiovallejo) [reply app/views/api/v3/analytics/user_analytics/_progress_cohort_report.html.erb:1]: That's right! I pulled this into a component to make it easy to delete :)

**Pull request merged**